src/core/Traits/EntityOpportunityDuplicator.php

```diff
@@ -0,0 +1,284 @@
+        }
+
+        if (isset($publishDate)) {
+            $phases = $app->repo('Opportunity')->findBy([
+                'parent' => $this->entityNewOpportunity
+            ]);
+    
+            foreach ($phases as $phase) {
+                if ($phase->getMetadata('isLastPhase')) {
+                    $phase->setPublishTimestamp($publishDate);
+                    $phase->save(true);
+                }
+            }
+        }       
+    }
+
+    private function duplicateMetadata() : void
+    {
+        foreach ($this->entityOpportunity->getMetadata() as $metadataKey => $metadataValue) {
+            if (!is_null($metadataValue) && $metadataValue != '') {
+                $this->entityNewOpportunity->setMetadata($metadataKey, $metadataValue);
+            }
+        }
+
+        $this->entityNewOpportunity->setTerms(['area' => $this->entityOpportunity->terms['area']]);
+        $this->entityNewOpportunity->setTerms(['tag' => $this->entityOpportunity->terms['tag']]);
+        $this->entityNewOpportunity->saveTerms();
+    }
+   
+    private function duplicateRegistrationFieldsAndFiles() : void
+    {
+        foreach ($this->entityOpportunity->getRegistrationFieldConfigurations() as $registrationFieldConfiguration) {
+            $fieldConfiguration = clone $registrationFieldConfiguration;
+            $fieldConfiguration->setOwnerId($this->entityNewOpportunity->id);
+            $fieldConfiguration->save(true);
+        }
+
+        foreach ($this->entityOpportunity->getRegistrationFileConfigurations() as $registrationFileConfiguration) {
+            $fileConfiguration = clone $registrationFileConfiguration;
+            $fileConfiguration->setOwnerId($this->entityNewOpportunity->id);
+            $fileConfiguration->save(true);
+        }
+
+    }
+
+    private function duplicateMetalist() : void
+    {
+        foreach ($this->entityOpportunity->getMetaLists() as $metaList_) {
+            foreach ($metaList_ as $metaList__) {
+                $metalist = clone $metaList__;
+                $metalist->setOwner($this->entityNewOpportunity);
+            
+                $metalist->save(true);
+            }
+        }
+    }
+
+    private function duplicateFiles() : void
+    {
+        $app = App::i();        
+
+        $src = PUBLIC_PATH . 'files/opportunity/' . $this->entityOpportunity->id;
+        $dst = PUBLIC_PATH . 'files/opportunity/' . $this->entityNewOpportunity->id;
+        $this->copyDir($src, $dst);
+        
+        $conn = $app->em->getConnection();
+        $files = $conn->fetchAll("SELECT * FROM file WHERE object_id = {$this->entityOpportunity->id} ORDER BY id ASC");
+        foreach ($files as $file) {
+            if (is_null($file['parent_id'])) {
+                $parentId = null;
+            } else if (isset($futureParentId) && !is_null($file['parent_id'])) {
+                $parentId = $futureParentId;
+            } else {
+                throw new Exception('File parent_id is null or not exists');
```

Comment:
Quando o `parent_id` for nulo, o IF executará primeira condição, logo, neste ponto (linha 210), `parent_id` sempre vem preenchido e não podemos dizer que é nulo ou inexistente. Assim, sugiro a mensagem 'File parent_id unexpect'.

Me perdoe pelos vários comentário, mas você teria outro ponto de vista, @vicmagpac ?